### PR TITLE
Use undo notification when deleting a card or badge

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Use notification with undo button instead of confirm dialog when deleting a card or a badge.
This PR also fixes some issues with notifications manager :  
- Adding a new notification did not clear well the previous one
- The deleted notification wasn't always shown because the element that fired the event was deleted. It's fixed by adding `showToast` method on the lovelace object.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by marcinbauer85** [comment]
Design documentation: https://www.figma.com/design/7QdMlBcwB8771cNKIsJ3ne/HA-02?node-id=2001-236&node-type=section&t=be0iHtGXLvcZtniH-11

File: src/panels/lovelace/editor/delete-badge.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    };
+
+    lovelace.showToast({
+      message: hass.localize("ui.common.successfully_deleted"),
+      duration: 8000,
+      action: { action, text: hass.localize("ui.common.undo") },
+    });
+  } catch (err: any) {
+    lovelace.showToast({
+      message: `Deleting failed: ${err.message}`,
```
Comment (wendevlin) [inline src/panels/lovelace/editor/delete-badge.ts:34]: Please add translation

File: src/panels/lovelace/editor/delete-card.ts [hunk 2]
```diff
@@ -1,44 +1,37 @@
...
+    };
+
+    lovelace.showToast({
+      message: hass.localize("ui.common.successfully_deleted"),
+      duration: 8000,
+      action: { action, text: hass.localize("ui.common.undo") },
+    });
+  } catch (err: any) {
+    lovelace.showToast({
+      message: `Deleting failed: ${err.message}`,
```
Comment (wendevlin) [inline src/panels/lovelace/editor/delete-card.ts:34]: Please add translation

File: src/panels/lovelace/components/hui-badge-edit-mode.ts [hunk 3]
```diff
@@ -231,8 +231,8 @@ export class HuiBadgeEditMode extends LitElement {
     fireEvent(this, "ll-edit-badge", { path: this.path! });
   }
 
-  private _deleteBadge(): void {
-    fireEvent(this, "ll-delete-badge", { path: this.path! });
+  private _deleteBadge({ silent }: { silent: boolean }): void {
```
Comment (wendevlin) [inline src/panels/lovelace/components/hui-badge-edit-mode.ts:234]: Why not using here a boolean too, like in DeleteBadgeParams?

**Review by wendevlin — CHANGES_REQUESTED** [review]

File: src/panels/lovelace/components/hui-badge-edit-mode.ts (hunk 3 above)
Comment (piitaya) [reply src/panels/lovelace/components/hui-badge-edit-mode.ts]: I simplified the implementation in the last commit. 👍

File: src/panels/lovelace/editor/delete-badge.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { HomeAssistant } from "../../../types";
```
Comment (wendevlin) [inline src/panels/lovelace/editor/delete-badge.ts:1]:
```suggestion
import type { HomeAssistant } from "../../../types";
```

File: src/panels/lovelace/editor/delete-badge.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+import { Lovelace } from "../types";
+import { deleteBadge } from "./config-util";
+import { LovelaceCardPath } from "./lovelace-path";
```
Comment (wendevlin) [inline src/panels/lovelace/editor/delete-badge.ts:4]:
```suggestion
import type { LovelaceCardPath } from "./lovelace-path";
```

File: src/panels/lovelace/editor/delete-card.ts [hunk 6]
```diff
@@ -1,44 +1,39 @@
...
-import { showDeleteCardDialog } from "./card-editor/show-delete-card-dialog";
-import { deleteCard, insertCard } from "./config-util";
-import {
-  LovelaceCardPath,
-  findLovelaceContainer,
-  getLovelaceContainerPath,
-  parseLovelaceCardPath,
-} from "./lovelace-path";
+import { deleteCard } from "./config-util";
+import { LovelaceCardPath } from "./lovelace-path";
```
Comment (wendevlin) [inline src/panels/lovelace/editor/delete-card.ts:4]:
```suggestion
import type { LovelaceCardPath } from "./lovelace-path";
```

File: src/panels/lovelace/ha-panel-lovelace.ts [hunk 7]
```diff
@@ -23,10 +23,11 @@ import { fetchResources } from "../../data/lovelace/resource";
 import { WindowWithPreloads } from "../../data/preloads";
 import "../../layouts/hass-error-screen";
 import "../../layouts/hass-loading-screen";
+import { ShowToastParams } from "../../managers/notification-manager";
```
Comment (wendevlin) [inline src/panels/lovelace/ha-panel-lovelace.ts:26]:
```suggestion
import type { ShowToastParams } from "../../managers/notification-manager";
```

File: src/panels/lovelace/types.ts [hunk 8]
```diff
@@ -14,6 +14,7 @@ import { LovelaceHeaderFooterConfig } from "./header-footer/types";
 import { LovelaceCardFeatureConfig } from "./card-features/types";
 import { LovelaceElement, LovelaceElementConfig } from "./elements/types";
 import { LovelaceHeadingBadgeConfig } from "./heading-badges/types";
+import { ShowToastParams } from "../../managers/notification-manager";
```
Comment (wendevlin) [inline src/panels/lovelace/types.ts:17]:
```suggestion
import type { ShowToastParams } from "../../managers/notification-manager";
```

File: src/panels/lovelace/views/hui-view.ts [hunk 9]
```diff
@@ -25,8 +25,8 @@ import { showCreateBadgeDialog } from "../editor/badge-editor/show-create-badge-
 import { showEditBadgeDialog } from "../editor/badge-editor/show-edit-badge-dialog";
 import { showCreateCardDialog } from "../editor/card-editor/show-create-card-dialog";
 import { showEditCardDialog } from "../editor/card-editor/show-edit-card-dialog";
-import { deleteBadge, deleteCard } from "../editor/config-util";
-import { confDeleteCard } from "../editor/delete-card";
+import { DeleteBadgeParams, performDeleteBadge } from "../editor/delete-badge";
```
Comment (wendevlin) [inline src/panels/lovelace/views/hui-view.ts:28]:
```suggestion
import { type DeleteBadgeParams, performDeleteBadge } from "../editor/delete-badge";
```

File: src/panels/lovelace/views/hui-view.ts [hunk 10]
```diff
@@ -25,8 +25,8 @@ import { showCreateBadgeDialog } from "../editor/badge-editor/show-create-badge-
...
+import { DeleteCardParams, performDeleteCard } from "../editor/delete-card";
```
Comment (wendevlin) [inline src/panels/lovelace/views/hui-view.ts:29]:
```suggestion
import { type DeleteCardParams, performDeleteCard } from "../editor/delete-card";
```

**Review by wendevlin — CHANGES_REQUESTED** [review]
I can check if we can add the type import check into our linter.

**Review by wendevlin — APPROVED** [review]

**Pull request merged**